src/core_ops.rs

```diff
@@ -4,13 +4,17 @@ use crate::node::Node;
 use crate::op_utils::{atom, check_arg_count};
 use crate::reduction::{Reduction, Response};
 
-const FIRST_COST: Cost = 1;
-const IF_COST: Cost = 1;
-const CONS_COST: Cost = 3;
-const REST_COST: Cost = 1;
-const LISTP_COST: Cost = 1;
-const CMP_BASE_COST: Cost = 2;
-const CMP_COST_PER_BYTE_DIVIDER: Cost = 64;
+const FIRST_COST: Cost = 30;
+// If cost lowered from 125 since it does not allocate anything
+const IF_COST: Cost = 75;
+// Cons cost lowered from 245. It only allocates a pair, which is small
+const CONS_COST: Cost = 80;
+// Rest cost lowered from 123 since it doesn't allocate anything
+const REST_COST: Cost = 65;
```

Comment:
How could first and rest possibly be different?

Reply:
I was asking myself the same thing when I edited this. I will take a deeper look at the benchmarks and results tomorrow

Reply:
It could be cache hits are more likely to come from `f` than `r`, but that would just be an artifact based on how programs tend to be formed, and wouldn't affect worst case, which is what we care about.

Reply:
rerunning these tests, it turned out to just be noise. I'll set them both to the same cost and probably lower it a bit compared to the measurement too